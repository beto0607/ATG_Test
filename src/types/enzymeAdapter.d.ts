// To avoid a typescript error with enzyme-adaoter-react-16
declare module 'enzyme-adapter-react-16';