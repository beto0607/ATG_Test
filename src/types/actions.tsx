import { GameSchedule, GameData } from "../types";

export const GAME_SCHEDULE_LOADED = "GAME_SCHEDULE_LOADED";
export type GAME_SCHEDULE_LOADED = typeof GAME_SCHEDULE_LOADED;

export const GAME_DATA_LOADED = "GAME_DATA_LOADED";
export type GAME_DATA_LOADED = typeof GAME_DATA_LOADED;


export interface GameScheduleLoadedAction {
    type: typeof GAME_SCHEDULE_LOADED;
    text: string;
    gameSchedule?: GameScheduleÂ | null;
}

export interface GameDataLoadedAction {
    type: typeof GAME_DATA_LOADED;
    gameData?: GameData;
}

export type LoadedActions = GameScheduleLoadedAction | GameDataLoadedAction;

export type AppActions = LoadedActions;