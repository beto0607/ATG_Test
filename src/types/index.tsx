export type GameInfo = {
    id: string,
    startTime: string // YYYY-MM-DDThh:mm:ss
}
//ENDPOINT RESPONSE
//https://www.atg.se/services/racinginfo/v1/api/products/<gameType>
export type GameSchedule = {
    betType: string;
    upcoming: Array<GameInfo>;
    results: Array<GameInfo>;
};
export type Person = {
    firstName: string,
    lastName: string
}
export type Horse = {
    name: string,
    trainer: Person,
    pedigree: {
        father: {
            name: string
        }
    }
}
export type RaceStart = {
    number: number,
    driver: Person,
    horse: Horse
}
export type Race = {
    id: string;
    name?: string;
    number: number;
    scheduledStartTime: string;
    starts: Array<RaceStart>;
}
//ENDPOINT RESPONSE
// https://www.atg.se/services/racinginfo/v1/api/games/<gameId>
export type GameData = {
    id: string,
    status: string,
    races: Array<Race>
};

export interface ApplicationState {
    text: string;
    gameSchedule: GameSchedule;
    gameData: Array<GameData>;
}