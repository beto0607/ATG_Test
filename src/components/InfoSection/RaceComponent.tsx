import React, { useState } from "react";
import { Collapse, Badge, Alert } from "react-bootstrap";
import { Race, RaceStart } from '../../types/index';
import Moment from "react-moment";
import styles from './race.module.scss';

export const StartComponent: React.FC<RaceStart> = ({ number, horse, driver }: RaceStart) => {
    const [open, setOpen] = useState(false);
    return (
        <div onClick={() => setOpen(!open)} className={styles['start-container']}>
            <div className={styles['start-title-container']}>
                <Badge variant="info">Start: #{number}</Badge>
                <span>Horse: <strong>{horse.name}</strong></span>
                <span>Driver: <strong>{`${driver.firstName} ${driver.lastName}`}</strong></span>
            </div>
            <Collapse in={open}>
                <div>
                    <span>Trainer: {`${horse.trainer.firstName} ${horse.trainer.lastName}`}</span>
                    <span>Horse father: {`${horse.pedigree.father.name}`}</span>
                </div>
            </Collapse>
        </div>
    )
};
export const RaceComponent: React.FC<Race> = ({ id, name, number, scheduledStartTime, starts }: Race) => {
    return (
        <div className={styles['race-container']}>
            <div className={styles['race-title-container']}>
                <Alert variant='info'>
                    <span>#{number} {name && <strong>- {name}</strong>}</span>
                </Alert>
                <Alert variant='info'>
                    Start time:
                    <Moment date={scheduledStartTime} format='HH:mm:ss' />
                </Alert>
            </div>
            {starts.map(start => <StartComponent key={`${id}_${start.number}`} {...start} />)}
        </div>
    );
};