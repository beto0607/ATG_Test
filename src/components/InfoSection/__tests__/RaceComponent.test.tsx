import React from 'react';
import { shallow } from 'enzyme';
import { StartComponent, RaceComponent } from '../RaceComponent';
import toJson from 'enzyme-to-json'
import { Race, RaceStart, Person, Horse } from '../../../types/index';
import { Collapse } from 'react-bootstrap';

const mockDriver: Person = {
    firstName: 'driver_first_name',
    lastName: 'driver_last_name'
};
const mockTrainer: Person = {
    firstName: 'trainer_name',
    lastName: 'trainer_last_name'
};
const mockHorse: Horse = {
    name: 'horse_one',
    trainer: mockTrainer,
    pedigree: { father: { name: 'father_horse' } }
}
const mockStart: RaceStart = {
    number: 1,
    driver: mockDriver,
    horse: mockHorse
}
const mockRace: Race = {
    id: 'an_id',
    name: 'a_name',
    number: 1,
    scheduledStartTime: "2019-07-24T11:20:04.492Z",
    starts: []
};
describe('StartComponent', () => {
    it('renders without crashing', () => {
        const props: RaceStart = mockStart;
        const component = shallow((<StartComponent {...props} />));
        expect(toJson(component)).toMatchSnapshot()
    });
    it('renders collapsed by default', () => {
        const props: RaceStart = mockStart;
        const component = shallow((<StartComponent {...props} />));
        expect(component.find(Collapse).prop('in')).toEqual(false);
    });
    it('renders expands when click', () => {
        const props: RaceStart = mockStart;
        const component = shallow((<StartComponent {...props} />));
        component.simulate('click');
        expect(component.find(Collapse).prop('in')).toEqual(true);
    });
});

describe('RaceComponent', () => {
    it('renders without crashing', () => {
        const props: Race = mockRace;
        const component = shallow((<RaceComponent {...props} />));
        expect(toJson(component)).toMatchSnapshot()
    });
    it('renders without crashing, name can be empty', () => {
        const props: Race = {...mockRace, name:''};
        const component = shallow((<RaceComponent {...props} />));
        expect(toJson(component)).toMatchSnapshot()
    });
    it('renders exactly 1 RaceStart', () => {
        const props: Race = { ...mockRace, starts: [mockStart] };
        const component = shallow((<RaceComponent {...props} />));
        expect(component.find(StartComponent)).toHaveLength(1)
    });
});

