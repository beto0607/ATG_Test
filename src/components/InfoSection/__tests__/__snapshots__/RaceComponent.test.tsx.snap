// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceComponent renders without crashing 1`] = `
<div
  className="race-container"
>
  <div
    className="race-title-container"
  >
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="info"
    >
      <span>
        #
        1
         
        <strong>
          - 
          a_name
        </strong>
      </span>
    </Alert>
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="info"
    >
      Start time:
      <t
        ago={false}
        calendar={false}
        date="2019-07-24T11:20:04.492Z"
        decimal={false}
        element={null}
        filter={[Function]}
        format="HH:mm:ss"
        fromNow={false}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      />
    </Alert>
  </div>
</div>
`;

exports[`RaceComponent renders without crashing, name can be empty 1`] = `
<div
  className="race-container"
>
  <div
    className="race-title-container"
  >
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="info"
    >
      <span>
        #
        1
         
      </span>
    </Alert>
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="info"
    >
      Start time:
      <t
        ago={false}
        calendar={false}
        date="2019-07-24T11:20:04.492Z"
        decimal={false}
        element={null}
        filter={[Function]}
        format="HH:mm:ss"
        fromNow={false}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      />
    </Alert>
  </div>
</div>
`;

exports[`StartComponent renders without crashing 1`] = `
<div
  className="start-container"
  onClick={[Function]}
>
  <div
    className="start-title-container"
  >
    <Badge
      pill={false}
      variant="info"
    >
      Start: #
      1
    </Badge>
    <span>
      Horse: 
      <strong>
        horse_one
      </strong>
    </span>
    <span>
      Driver: 
      <strong>
        driver_first_name driver_last_name
      </strong>
    </span>
  </div>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div>
      <span>
        Trainer: 
        trainer_name trainer_last_name
      </span>
      <span>
        Horse father: 
        father_horse
      </span>
    </div>
  </Collapse>
</div>
`;
